docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -150,7 +150,180 @@ This is then used to populate the template fields in our [Dockerrun.aws.json](ht
 
 The Layer0 configuration referencing the AWS DynamoDB configuration `table_name = "${aws_DynamoDB_table.guestbook.name}"`, infers an implicit dependency. Before Terraform creates the infrastructre, it will use this information to order the resources created and also create resources in parallel where there are no dependencies. In this example, the AWS DynamoDB table will be created before the Layer0 deploy. See [Terraform Dependencies](https://www.terraform.io/intro/getting-started/dependencies.html) for more information.
 
-## Part 4: Terraform Destroy
+## Part 4: Terraform Remote State
+
+Terraform stores the state of the deployed infrastructure in a local file named `terraform.tfstate` by default. This includes not only information about the resources deployed but also metadata such as resource dependencies. To find out more about why Terraform needs to store state see [Purpose of Terraform State](https://www.terraform.io/docs/state/purpose.html). 
+
+How state is loaded and used for operations such as `terraform apply` is determined by a [Backend](https://www.terraform.io/docs/backends). As mentioned, by default the state is stored locally which is enabled by a "local" backend.
+
+### Remote State
+In scenarios where you are working as part of a team to provision and manage a services deployed by Terraform, all the members of the team will need access to the state file to apply new changes. You would also want resiliency against losing the state file. You can cater for provisioning and managing Terraform in a team environment and providing redundancy for the state file by using a remote backend. Backends such as [Consul](https://www.terraform.io/docs/backends/types/consul.html) & [S3](https://www.terraform.io/docs/backends/types/s3.html) among others, are backends that you can use to store master Terraform state in a remote location.
+
+To configure a remote backend, append the `terraform` section below to your terraform file `layer0.tf`. Update the `path` property to a GUID to avoid a potential conflict as demo.consul.io is a public consul endpoint.
+
+```
+terraform {
+  backend "consul" {
+    address = "demo.consul.io"
+    path    = "a-unique-random-string"
+    lock    = false
+  }
+}
+```
+
+Once you have modified `layer0.tf`, you will need to initailize the newly configured backend by running the following command.
```

Comment:
This section is pretty verbose - I assume it's covered pretty well by the terraform docs. I don't feel too strongly either way, but would consider removing chunks in favor of links to terraform docs.